middleware/middleware.go

```diff
@@ -2,14 +2,21 @@ package middleware
 
 import (
 	"context"
+	"google.golang.org/grpc"
 )
 
 // Handler defines the handler invoked by Middleware.
 type Handler func(ctx context.Context, req interface{}) (interface{}, error)
 
+// StreamHandler defines the handler invoked by Middleware for stream calls.
+type StreamHandler func(srv interface{}, stream grpc.ServerStream) error
+
 // Middleware is HTTP/gRPC transport middleware.
 type Middleware func(Handler) Handler
 
+// StreamMiddleware is gRPC stream transport middleware.
+type StreamMiddleware func(StreamHandler) StreamHandler
```

Comment:
Can all of these be deleted

Reply:
removed

Reply:
It looks much simpler now, can you add some individual tests?

Reply:
sure, I'll add tests